cli/bench/bench.go

```diff
@@ -0,0 +1,110 @@
+package bench
+
+import (
+	"fmt"
+	"sync"
+	"time"
+
+	"github.com/FZambia/tarantool"
+	"github.com/tarantool/cartridge-cli/cli/context"
+)
+
+func space_preset(ctx context.BenchCtx, tarantool_connection *tarantool.Connection, space_name string) error {
+	drop_space(ctx, tarantool_connection, space_name)
+	err := create_space(tarantool_connection, space_name)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+func count_request(err error, results *Results) {
+	if err == nil {
+		results.success_result++
+	} else {
+		results.failed_result++
+		fmt.Println(err)
+		panic(err)
+	}
+	results.hanled_requests_count++
+}
+
+func connection_loop(ctx context.BenchCtx, connection_id int, requests_loop request_t, space_name string, results *Results, wait_group *sync.WaitGroup, bench_active_flag *bool) {
+	tarantool_connection, err := tarantool_connect(ctx)
+	if err != nil {
+		return
+	}
+	defer tarantool_connection.Close()
+
+	var connection_wait sync.WaitGroup
+	for request_id := 0; request_id < ctx.SimultaneousRequests; request_id++ {
+		connection_wait.Add(1)
+		go requests_loop(ctx, connection_id, request_id, tarantool_connection, space_name, results, &connection_wait, bench_active_flag)
+	}
+	connection_wait.Wait()
+	defer wait_group.Done()
+}
+
+func Run(ctx context.BenchCtx) error {
+	max_connections_count := ctx.Connections
+
+	bench_active_flag := true
+
+	timer_duration := ctx.Duration
+
+	operations_map := []Operaiton{
+		{"insert", ctx.InsertCount, insert_requests_loop, 0, 0},
+		{"delete", ctx.DeleteCount, delete_requests_loop, 0, 0},
+		{"update", ctx.UpdateCount, update_requests_loop, 0, 0},
+		{"select", ctx.SelectCount, select_requests_loop, 0, 0},
+	}
+
+	all_operation_count := 0
+	for _, operation := range operations_map {
+		all_operation_count += operation.count
+	}
+
+	for i := 0; i < len(operations_map); i++ {
+		operations_map[i].percent = float64(operations_map[i].count) / float64(all_operation_count)
+		operations_map[i].connections_count = int(float64(max_connections_count) * operations_map[i].percent)
+		if operations_map[i].count != 0 && operations_map[i].connections_count == 0 {
+			operations_map[i].connections_count = 1
+		}
+	}
```

Comment:
Can you itarate over operations_map via ``range``?

Reply:
Unfortunately, range returns by value, i need by reference

Reply:
```go
for i, _ := range operations_map {
... operations_map[i] = ...
}
```